Assets/Scripts/DetectLookedAtInteractive.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Detects interactive elements player is looking at
+/// https://docs.unity3d.com/ScriptReference/Physics.Raycast.html
+/// </summary>
+
+public class DetectLookedAtInteractive : MonoBehaviour
+{
+    [Tooltip("Starting point of raycast used to detect interactives.")]
+    [SerializeField]
+    private Transform raycastOrigin;
+    [Tooltip("How far from the raycast origin we will search for interactive elements")]
+    [SerializeField]
+    private float maxRange = 5.0f;
+
+    /// <summary>
+    /// Event raised when the player looks at a different IInteractive.
+    /// </summary>
+    public static event Action<IInteractive> LookedAtInteractiveChanged;
+
+    public IInteractive LookedAtInteractive
```

Comment:
I would comment this out as well, help describe what it does.

Reply:
Done.